neptune/new/sync/utils.py

```diff
@@ -0,0 +1,178 @@
+    "To synchronize offline run, specify the project name with the --project flag "
+    "or by setting the {} environment variable.".format(PROJECT_ENV_NAME)
+)
+
+
+def _project_not_found_message(project_name: QualifiedName) -> str:
+    return (
+        "Project {} not found. Could not synchronize offline runs. ".format(
+            project_name
+        )
+        + "Please ensure you specified the correct project name with the --project flag "
+        + "or with the {} environment variable, or contact Neptune for support.".format(
+            PROJECT_ENV_NAME
+        )
+    )
+
+
+def get_project(
+    project_name_flag: Optional[QualifiedName], backend: NeptuneBackend
+) -> Optional[Project]:
+    project_name = project_name_flag or QualifiedName(os.getenv(PROJECT_ENV_NAME))
+    if not project_name:
+        click.echo(textwrap.fill(_project_name_missing_message), file=sys.stderr)
+        return None
+    try:
+        return backend.get_project(project_name)
+    except ProjectNotFound:
+        click.echo(
+            textwrap.fill(_project_not_found_message(project_name)), file=sys.stderr
+        )
+        return None
+
+
+def get_qualified_name(run: ApiExperiment) -> str:
+    return "{}/{}/{}".format(run.workspace, run.project_name, run.sys_id)
+
+
+def is_valid_uuid(val: Any) -> bool:
+    try:
+        uuid.UUID(val)
+        return True
+    except ValueError:
+        return False
+
+
+def is_run_synced(run_path: Path) -> bool:
+    return all(
+        _is_execution_synced(execution_path) for execution_path in run_path.iterdir()
+    )
+
+
+def _is_execution_synced(execution_path: Path) -> bool:
+    disk_queue = DiskQueue(
+        execution_path,
+        lambda x: x.to_dict(),
+        Operation.from_dict,
+        threading.RLock(),
+    )
+    return disk_queue.is_empty()
+
+
+def split_dir_name(dir_name: str) -> Tuple[ContainerType, UniqueId]:
+    parts = dir_name.split("__")
+    if len(parts) == 2:
+        return ContainerType(parts[0]), UniqueId(parts[1])
+    elif len(parts) == 1:
+        return ContainerType.RUN, UniqueId(dir_name)
+    else:
+        raise ValueError(f"Wrong dir format: {dir_name}")
+
+
+def iterate_containers(
+    base_path: Path,
+) -> Iterator[Tuple[ContainerType, UniqueId, Path]]:
+    if not base_path.is_dir():
+        return
+
+    for path in base_path.iterdir():
+        container_type, unique_id = split_dir_name(dir_name=path.name)
+
+        yield container_type, unique_id, path
+
+
+def get_offline_dirs(base_path: Path) -> List[UniqueId]:
+    result = []
+    if not (base_path / OFFLINE_DIRECTORY).is_dir():
+        return []
+    for path_ in (base_path / OFFLINE_DIRECTORY).iterdir():
+        dir_name = path_.name
+        result.append(UniqueId(dir_name))
+    return result
+
+
+def is_offline_run_name(name: str) -> bool:
+    if not name.startswith(OFFLINE_NAME_PREFIX):
+        return False
+
+    # just make sure that dir has accepted format or raise exception
```

Comment:
err, what? this function shouldn't raise -- it should return `True/False`

Reply:
Ok

Reply:
Ok, right.
I've also changed script behaviour.

If user passes wrong offline path, he gets info that such run is not found, instead of info about wrong format.